pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -1258,6 +1258,24 @@ public LocalDateTime toLocalDateTimeBin(byte[] bytes) throws PSQLException {
     return LocalDateTime.ofEpochSecond(parsedTimestamp.millis / 1000L, parsedTimestamp.nanos, ZoneOffset.UTC);
   }
 
+  /**
+   * Returns the local date time object matching the given bytes with {@link Oid#TIMESTAMP} or
+   * {@link Oid#TIMESTAMPTZ}.
+   * @param bytes The binary encoded local date time value.
+   *
+   * @return The parsed local date time object.
+   * @throws PSQLException If binary format could not be parsed.
+   */
+  public LocalDate toLocalDateBin(byte[] bytes) throws PSQLException {
+	  if (bytes.length != 4) {
+	    throw new PSQLException(GT.tr("Unsupported binary encoding of {0}.", "date"),
+	        PSQLState.BAD_DATETIME_FORMAT);
+	  }
+	  int days = ByteConverter.int4(bytes, 0);
+	  days += 10957L; // adapt the different Postgres Epoch
+	  return LocalDate.ofEpochDay(days);
```

Comment:
Here I am not fully sure if we need the min/max checks. But Postgres Dates should always fit into LocalDate, so the check is IMHO obsolete. What do others think?

Reply:
Check `org.postgresql.test.jdbc42.GetObject310InfinityTests` and https://www.postgresql.org/docs/9.0/datatype-datetime.html#AEN5647

I wonder how would you fit `-infinity` and `infinity` into `LocalDate` :)

Reply:
Ok, PostgreSQL defines `-infinity` and `infinity` as MIN_INT32 and MAX_INT32: https://github.com/postgres/postgres/blob/27b77ecf9f4d5be211900eda54d8155ada50d696/src/include/utils/date.h#L34-L43
So wire format is exactly that: "min_int" vs "max_int"

Reply:
Apparently, `days += 10957L` would overflow `int32`.

So I suggest treating `MAX_INT` and `MIN_INT` as a special case, and use `long` variable for `days`.
It is really sad that Java allows integer overflow in `days += 10957L`

Reply:
Will add a test and implement this check.

Reply:
The test already exists, no need to add it.
However, it would probably be interesting to test cases like "infinity - 1 day" (e.g. max_int32 - 1day).

Reply:
OK, I fixed the overflow. Actually LocalDate.fromEpochDays takes a long, I just declared the variable in a wrong way (copynpaste - shitnwaste),
I also added the Integer.MIN_VALUE and Integer.MAX_VALUE checks in the same way like for LocalDateTime.

Reply:
The `GetObject310InfinityTests` now also passes.